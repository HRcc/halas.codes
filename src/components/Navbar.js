import React from 'react'
import PropTypes from 'prop-types'
import logo from '../assets/images/logo.svg'
import './Navbar.css'

const Navbar = ({ links }) => (
  <nav className="h-16 px-4 flex justify-between items-center">
    <img src={logo} alt="logo" className="h-12"/>

    {links && <ul className="list-reset flex">
      {links.map(link => (
        <li key={link.label}>
          <a href={link.href} target="_blank" className="px-4 text-xs flex items-center font-hairline no-underline uppercase text-grey-light hover:text-white">
            <i className={link.icon + ' text-base mr-2 text-shadow2'}/>
            {link.label}
          </a>
        </li>
      ))}
    </ul>}
  </nav>
)

Navbar.propTypes = {
  links: PropTypes.array
}

export default Navbar
