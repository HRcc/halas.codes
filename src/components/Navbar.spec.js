import React from 'react'
import { shallow } from 'enzyme'
import Navbar from './Navbar'

it('renders navbar without links', () => {
  const wrapper = shallow(<Navbar />)

  expect(wrapper).toMatchSnapshot()
})

it('renders navbar with links and icons', () => {
  const links = [
    {
      href: 'github.url',
      icon: 'fab fa-github',
      label: 'Github'
    },
    {
      href: 'linkedin.url',
      icon: 'fab fa-linkedin',
      label: 'LinkedIn'
    }
  ]
  const wrapper = shallow(<Navbar links={links}/>)

  expect(wrapper).toMatchSnapshot()
})
