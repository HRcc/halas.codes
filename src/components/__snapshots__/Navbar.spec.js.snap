// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar with links and icons 1`] = `
<nav
  className="h-16 px-4 flex justify-between items-center"
>
  <img
    alt="logo"
    className="h-12"
    src="logo.svg"
  />
  <ul
    className="list-reset flex"
  >
    <li
      key="Github"
    >
      <a
        className="px-4 text-xs flex items-center font-hairline no-underline uppercase text-grey-light hover:text-white"
        href="github.url"
        target="_blank"
      >
        <i
          className="fab fa-github text-base mr-2 text-shadow2"
        />
        Github
      </a>
    </li>
    <li
      key="LinkedIn"
    >
      <a
        className="px-4 text-xs flex items-center font-hairline no-underline uppercase text-grey-light hover:text-white"
        href="linkedin.url"
        target="_blank"
      >
        <i
          className="fab fa-linkedin text-base mr-2 text-shadow2"
        />
        LinkedIn
      </a>
    </li>
  </ul>
</nav>
`;

exports[`renders navbar without links 1`] = `
<nav
  className="h-16 px-4 flex justify-between items-center"
>
  <img
    alt="logo"
    className="h-12"
    src="logo.svg"
  />
</nav>
`;
